src/stores/GlobalStore.js

```diff
@@ -74,33 +74,53 @@ export default class GlobalStore {
    */
   registerQrypto = () => {
     if (!this.localWallet) {
-      console.log('Registering with Qrypto...'); // eslint-disable-line
-      window.addEventListener('message', this.handleQryptoAccountChange, false);
+      console.log('Trying to register with Qrypto...'); // eslint-disable-line
+      window.addEventListener('message', this.handleWindowMessage, false);
       window.postMessage({ message: { type: 'CONNECT_QRYPTO' } }, '*');
     }
   }
 
+  /**
+   * Handles all window messages.
+   * @param {MessageEvent} event Message to handle.
+   */
+  handleWindowMessage = (event) => {
+    if (event.data.message && event.data.message.type) {
+      const types = {
+        QRYPTO_INSTALLED_OR_UPDATED: this.handleQryptoInstall,
+        ACCOUNT_CHANGED: this.handleQryptoAccountChange,
+      };
+      const messageAction = types[event.data.message.type];
+      if (messageAction) messageAction(event);
```

Comment:
what does this do?

Reply:
This checks the `window message event` and if any of them are `QRYPTO_INSTALLED_OR_UPDATED` or `ACCOUNT_CHANGED` it will execute their functions and pass the `event` param along with it.

Reply:
```
// so you have your types. these correspond to a function in the class:
const types = {
        QRYPTO_INSTALLED_OR_UPDATED: this.handleQryptoInstall,
        ACCOUNT_CHANGED: this.handleQryptoAccountChange,
      };

// this fetches the proper function to execute, e.g. types[QRYPTO_INSTALLED_OR_UPDATED] = this.handleQryptoInstall. messageAction is now pointing the the function.
const messageAction = types[event.data.message.type];

// if messageAction is a defined function, execute the function: messageAction(event) === this.handleQryptoInstall(event)
if (messageAction) messageAction(event);
```